drivers/led_strip/ws2812_gpio.c

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/led_strip.h>
+
+#include <string.h>
+
+#define LOG_LEVEL CONFIG_LED_STRIP_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ws2812_gpio);
+
+#include <zephyr.h>
+#include <soc.h>
+#include <drivers/gpio.h>
+#include <device.h>
+#include <drivers/clock_control.h>
+
+#define WS2812_DEV          DT_INST_0_WORLDSEMI_WS2812_GPIO_IN_GPIOS_CONTROLLER
+#define WS2812_PIN          DT_INST_0_WORLDSEMI_WS2812_GPIO_IN_GPIOS_PIN
+#define WS2812_PIN_FLAGS    DT_INST_0_WORLDSEMI_WS2812_GPIO_IN_GPIOS_FLAGS
+
+#define RED_ORDER    DT_INST_0_WORLDSEMI_WS2812_GPIO_RED_ORDER
+#define GREEN_ORDER  DT_INST_0_WORLDSEMI_WS2812_GPIO_GREEN_ORDER
+#define BLUE_ORDER   DT_INST_0_WORLDSEMI_WS2812_GPIO_BLUE_ORDER
+#define WHITE_ORDER  DT_INST_0_WORLDSEMI_WS2812_GPIO_WHITE_ORDER
+
+#if (DT_INST_0_WORLDSEMI_WS2812_SPI_HAS_WHITE_CHANNEL == 1)
+#define HAS_WHITE 1
```

Comment:
Couldn't it be written simply as
```
#define HAS_WHITE DT_INST_0_WORLDSEMI_WS2812_SPI_HAS_WHITE_CHANNEL
```
I guess the concern is that `DT_INST_0_WORLDSEMI_WS2812_SPI_HAS_WHITE_CHANNEL` may be undefined. But even if this is the case the `#if HAS_WHITE` will work. Otherwise this define is generated from `has-white-channel` property of type binary.

Reply:
You're right, thanks.